go/mysql/fakesqldb/server.go

```diff
@@ -347,15 +349,22 @@ func (db *DB) HandleQuery(c *mysql.Conn, query string, callback func(*sqltypes.R
 	// Check if we should close the connection and provoke errno 2013.
 	if db.shouldClose {
 		c.Close()
-		callback(&sqltypes.Result{})
+
+		//log error
+		if err := callback(&sqltypes.Result{}); err != nil {
+			log.Errorf("callback(&sqltypes.Result{}) failed : %v", err)
+		}
 		return nil
 	}
 
 	// Using special handling for 'SET NAMES utf8'.  The driver
 	// may send this at connection time, and we don't want it to
 	// interfere.
 	if key == "set names utf8" {
-		callback(&sqltypes.Result{})
+		//log error
+		if err2 := callback(&sqltypes.Result{}); err2 != nil {
```

Comment:
```suggestion
		if err := callback(&sqltypes.Result{}); err != nil {
```

Reply:
The scope of `err` here is just within the `if` statement. Unless there is already a variable with the same name and a larger scope, it should be fine to reuse `err`.